cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/CoreSchedulerService.java

```diff
@@ -431,6 +431,18 @@ public Void run(ProgramScheduleStoreDataset store, JobQueueDataset queue) {
   }
 
   @Override
+  public void modifySchedulesTriggeredByDeletedProgram(final ProgramId programId) {
+    checkStarted();
+    execute(new StoreAndQueueTxRunnable<Void, RuntimeException>() {
+      @Override
+      public Void run(ProgramScheduleStoreDataset store, JobQueueDataset queue) {
+        store.modifySchedulesTriggeredByDeletedProgram(programId);
```

Comment:
Currently, the returned list of deleted schedules is not used, and the corresponding jobs in JobQueue are not marked for deletion. Therefore, if the job is in PENDING_TRIGGER state, it will be deleted when it expires, and if it's in PENDING_CONSTRAINT or PENDING_LAUNCH state, it will remain in the job queue to wait for launching. For updated schedule with composite trigger, it will be marked as OBSOLETE when new notification arrives and will be deleted. I think we need to consider whether we want to keep such behavior, since the user didn't explicitly delete the schedule, and it will be unexpected for users that deleting an application can affect the job's state of another application

Reply:
i don't understand this, what gets marked obsolete and deleted? how is that related to jobs in the queue?

Reply:
When a schedule is modified, the corresponding job in the queue is marked as OBSOLETE. Jobs marked as OBSOLETE will be deleted when it's reached.

Reply:
still don't understand how this is related to the pending states you mentioned in your comment. If a job is in any of those pending states and the schedule is modified, what happens? 

I would imagine that users expect that anything that was triggered before schedule modification will still be launched. Is this not the case?